experimental/devices/cap1188/cap1188_options.go

```diff
@@ -0,0 +1,140 @@
+package cap1188
+
+import "periph.io/x/periph/conn/gpio"
+
+// SamplingTime determines the time to take a single sample
+type SamplingTime uint8
+
+// Possible sampling time values (written as 2 bits)
+const (
+	S320us SamplingTime = 0
+	S640us SamplingTime = 1
+	// S1_28ms represents 1.28ms sampling time, which is the default.
+	S1_28ms SamplingTime = 2 // default
+	S2_56ms SamplingTime = 3
+)
+
+// AvgSampling set the number of samples per measurement that get
+// averaged
+type AvgSampling uint8
+
+// possible average sampling values. (written as 3 bits)
+const (
+	// Avg1 means that 1 sample is taken per measurement
+	Avg1   AvgSampling = iota // 0
+	Avg2                      // 1
+	Avg4                      // 2
+	Avg8                      // 3 default
+	Avg16                     // 4
+	Avg32                     // 5
+	Avg64                     // 6
+	Avg128                    // 7
+)
+
+// CycleTime determines the overall cycle time for all measured channels during
+// normal operation.
+type CycleTime uint8
+
+// possible cycle time values. (written as 2 bits)
+const (
+	C35ms CycleTime = iota // 0
+	C70ms                  // default
+	C105ms
+	C140ms
+)
+
+// MaxDur is the maximum duration of a touch event before it triggers a
+// recalibration.
+type MaxDur uint8
+
+// possible touch duration values. (written as 4 bits)
+const (
+	MaxDur560ms MaxDur = iota
+	MaxDur840ms
+	MaxDur1120ms
+	MaxDur1400ms
+	MaxDur1680ms
+	MaxDur2240ms
+	MaxDur2800ms
+	MaxDur3360ms
+	MaxDur3920ms
+	MaxDur44800ms
+	MaxDur5600ms // default
+	MaxDur6720ms
+	MaxDur7840ms
+	MaxDur8906ms
+	MaxDur10080ms
+	MaxDur11200ms
+)
+
+// Opts is optional options to pass to the constructor.
+//
+// Address is only used on creation of an I²C-device. Its default value is 0x28.
+// It can be set to other values (0x29, 0x2a, 0x2b, 0x2c) depending on the HW
+// configuration of the ADDR_COMM pin. This has no effect with NewSPI()
+type Opts struct {
+	// Address is the I2C slave address to use
+	Address uint16
+
+	// Debug turns on extra logging capabilities
+	Debug bool
+	// LinkedLEDs indicates if the LEDs should be activated automatically
```

Comment:
Oh since you noted that the functions are useful, then do not make it an option and have users call the function explicitly.

Reply:
options are embedded now, so it's all good, users can set the default behavior and change it as needed.